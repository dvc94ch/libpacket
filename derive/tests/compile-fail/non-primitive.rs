// Copyright (c) 2021 Pierre Chifflier <chifflier@wzdftpd.net>
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

use libpacket_derive::Packet;

#[derive(Clone, Debug)]
pub struct Toto{
    dummy: u16,
}

#[derive(Packet)]
pub struct PacketU16 {
    banana: Toto,
    #[payload]
    payload: Vec<u8>,
}

fn main() {}
