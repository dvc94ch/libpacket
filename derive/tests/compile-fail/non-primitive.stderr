error: non-primitive field types must specify #[construct_with]
  --> $DIR/non-primitive.rs:18:13
   |
18 |     banana: Toto,
   |             ^^^^
